### PR TITLE
:bug: Fix probable bug

### PR DESCRIPTION
**Pull request by cupakob**
Possible integer overflow when converting int -> uint16. This bug may never occur, but it is better to have proper handling of this case.

## Related issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/reconciler/topology/partitionset/partitionset_reconcile.go [hunk 1]
```diff
@@ -97,7 +99,11 @@ func (c *controller) reconcile(ctx context.Context, partitionSet *topologyv1alph
 	} else {
 		matchLabelsMap = partition(shards, dimensions, nil)
 	}
-	partitionSet.Status.Count = uint16(len(matchLabelsMap))
```
Comment (sttts) [inline pkg/reconciler/topology/partitionset/partitionset_reconcile.go:100]: can we just make this an uint?

Comment (cupakob) [reply pkg/reconciler/topology/partitionset/partitionset_reconcile.go:100]: should not be a problem...but then why not int? what is the advantage of the uint?

Comment (stevekuznetsov) [reply pkg/reconciler/topology/partitionset/partitionset_reconcile.go:100]: Negative numbers of shards doesn't make any sense either :)

Comment (cupakob) [reply pkg/reconciler/topology/partitionset/partitionset_reconcile.go:100]: yeah, sure...but then another convert is needed from uint to int. In this case `len()` should not return a negative number ;)

Comment (cupakob) [reply pkg/reconciler/topology/partitionset/partitionset_reconcile.go:100]: ok, changed to uint

**Review by embik — APPROVED** [review]
/approve

**Pull request merged**